src/map/ai/helpers/pathfind.cpp

```diff
@@ -401,6 +401,34 @@ void CPathFind::FollowPath(time_point tick)
     }
 }
 
+void CPathFind::SnapToCollision(position_t& point)
+{
+    auto zone = zoneutils::GetZone(m_POwner->getZone());
+    if (zone != nullptr)
+    {
+        position_t downTarget{ point.x, point.y - 100, point.z, 0, 0 };
+        position_t upTarget{ point.x, point.y + 100, point.z, 0, 0 };
```

Comment:
100 is a very long way, be careful with this both for correctness and performance. Test this out in Garliage, for instance

Reply:
Now that I think about it, seeking upward after you've sought down is a good pattern, and gets rid of my fear of snapping to the ceiling, but the 100 is still concerning